styx-service-common/src/test/java/com/spotify/styx/storage/DatastoreStorageTest.java

```diff
@@ -1131,8 +1105,13 @@ public void runInTransactionWithRetriesShouldPropagateDatastoreIOException() thr
   public void runInTransactionWithRetriesShouldThrowIfDatastoreNewTransactionFails() throws Exception {
     var cause = new DatastoreIOException(new DatastoreException(1, "", ""));
     doThrow(cause).when(datastore).newTransaction();
-    exception.expect(is(cause));
-    storage.runInTransactionWithRetries(transactionFunction);
+
+    var exception = assertThrows(
+        DatastoreIOException.class,
+        () -> storage.runInTransactionWithRetries(transactionFunction)
+    );
+
+    assertThat(exception, is(cause));
```

Comment:
these tests look a bit too rigid, but good refactoring

Reply:
This was more a "Boy Scout rule" change as the way for asserting for exception using `@Rule` is now deprecated in favor of `assertThrows`. Functionally this test is as strict as the old version, just using a not deprecated API

Reply:
yes I agree. I mean that the test itself, not the refactoring